apps/client/src/app/pages/portfolio/fire/fire-page.html

```diff
@@ -160,10 +169,28 @@ <h4 class="align-items-center d-flex m-0">
             <gf-premium-indicator class="ml-1" />
           }
         </h4>
-        <gf-rules
-          [hasPermissionToCreateOrder]="hasPermissionToCreateOrder"
-          [rules]="feeRules"
-        />
+        @if (!isLoadingPortfolioReport) {
+          <gf-rules
+            [hasPermissionToCreateOrder]="hasPermissionToCreateOrder"
+            [rules]="feeRules"
+            (rulesUpdated)="onRulesUpdated($event)"
+          />
+        }
+      </div>
+      <div>
```

Comment:
Would wrapping this `div` solve the issue described [here](https://github.com/ghostfolio/ghostfolio/pull/3537#issuecomment-2278197002)?

```
@if (inactiveRules?.length > 0) {
  <div>
    ...
  </div>
}
```

As it is the last block, I think it is not a big problem if it is suddenly visible (without skeleton).

Reply:
It wouldn't fix it, I was talking about the other categories. For example `Emergency Fund` is in constant loading if none of its rules are enabled, because of this: https://github.com/ghostfolio/ghostfolio/blob/2893d71377073ad6c53acd0230e5b0f9a2cad9d4/apps/client/src/app/components/rules/rules.component.html#L12.
I think a solution would be to check if the object containing the rules is of length 0 instead. The endpoint anyways return always the result for all rules in 1 response, so checking the individual length of rules to see if they are still loading is useless.

Reply:
I see. Maybe the rules per group could be initialized as `rules: string[] = undefined;`

Then check like this:

```
@if (!rules) {
  <gf-rule [isLoading]="true" />
} @else if (rules.length > 0) {
  @for (rule of rules; track rule) {
    <gf-rule [rule]="rule" />
  }
}
```

What do you think?

Reply:
I still think it is useless to have a check per rule set, because the backend returns all of them at the same time anyways. I'll push a commit that do it your way, but I think the way is to set this `isLoading` in the fire page, for all rules at once.

Reply:
I actually pushed a version that seems optimal, let me know what you think. The `initialiazePortfolioReport` lets the children component know when the report has finished loading.
Last thing is that if the `inactive` section does not have the same top padding compared to the other rules section, trying to figure out why.